### ISSUE TITLE
Typescript: Nested Generic Type Causes Syntax Error

### ISSUE THREAD

**Issue by mahdi-farnia** [prettier/prettier-vscode#2642]
### Summary

**This error does not happen in prettier playground**

**Extension Version**: 9.5.0
**VSCode Version**: 1.69.2
**Parser**: Typescript
**OS Version**: MacOS 12.4

### Github Repository to Reproduce Issue

[Link to a Github repo that can be used to reproduce the issue.](https://github.com/mahdi-farnia/PrettierExtensionError)

### Playground Link (Works Properly)
https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AE6AznAMowCG8APACoB8AFAG4UA2S+tAlJwNpcANPhace+ALz18zCDgAm+ALr5gAHSj4t+AE5wY6HZr6s2w0V26Tp6zdvunJMgNwb7AXyUb3GjTACeAA5w+ADy6DBs+kQAwtDwAB4w5FQhEvgASvqGULRBcNQBwRCohCQpNFDoALYARnA69PSuUCCCIBCBuNBEyKAUOjoQAO4ACgMIvSjswxT+ve21OhRgANb6pBTVcAAyOFBwyKjsJIvLaxuBK-sA5sgwOuhw7XB1cPLy7zsUUDfoFDc4AAxCA6apUXC-ZAgCgRCBtEAACxg1TYAHVETh4EQrmAyJMsThmFj-NCwEQFiB9iQdDBRssbuCjidniAAFZEBKkW5RACK6Ag8GZbFOICuOhp0KKcCIYB0OC6CMC8tgaIUMERyAAHAAGdrKiAkNHLQLQ5UyhrMQ7tACOAvg9M6UxhRAAtAd3u8EXo7Tg9PSAUykMcRaySNUcPdHmGeXB+YLDsGWe1KLU1fINcgAEwp5Y4Ni3OLVIMgGUAVgRxDgtAotSmIdFzCeAEkoJ9YKQ5QqYABBNvkfxRYUkdzuIA

### Steps To Reproduce:

1.  Create A Nested Type Like This:
```ts
type MyType = ReturnType<typeof useState<number>>;
```

### Expected result

Should Be Format File

### Actual result

`SyntaxError: '>' expected.`

## Prettier Log Output

```
at qe (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/node_modules/prettier/parser-typescript.js:1:15669)
    at UU (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/node_modules/prettier/parser-typescript.js:280:5918)
    at Object.zU [as parse] (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/node_modules/prettier/parser-typescript.js:280:6241)
    at Object.parse (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/node_modules/prettier/index.js:7330:23)
    at coreFormat (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/node_modules/prettier/index.js:8641:18)
    at formatWithCursor2 (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/node_modules/prettier/index.js:8833:18)
    at /Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/node_modules/prettier/index.js:39137:12
    at Object.Success [as format] (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/node_modules/prettier/index.js:39151:12)
    at t.default.format (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/src/PrettierEditService.ts:463:45)
    at t.PrettierEditProvider.provideEdits (/Users/mahdifarnia/.vscode/extensions/esbenp.prettier-vscode-9.5.0/src/PrettierEditService.ts:322:22)
    at F.provideDocumentFormattingEdits (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:87:45332)
["INFO" - 2:52:19 PM] Formatting completed in 0.009ms.
```

**Comment by jarpoole** [comment]
This is related to issue #2608 and fixed in the latest release ([9.6.0](https://github.com/prettier/prettier-vscode/releases/tag/v9.6.0)). Unfortunately this release hasn't been published to the vscode extension store so we just need to wait. @ntotten any chance you could resolve this one? I noticed that CI tried to automatically publish to vscode but errored out. Maybe it could be as simple as just running it again?

**Comment by mahdi-farnia** [comment]
> This is related to issue #2608 and fixed in the latest release ([9.6.0](https://github.com/prettier/prettier-vscode/releases/tag/v9.6.0)). Unfortunately this release hasn't been published to the vscode extension store so we just need to wait. @ntotten any chance you could resolve this one? I noticed that CI tried to automatically publish to vscode but errored out. Maybe it could be as simple as just running it again?
> 
> 

If i download the released vsix file (9.6.0) the error will gone?
